retry/integration_test.go

```diff
@@ -180,10 +184,32 @@ func (s *RetryTripperwareSuite) TestCustomResponseDiscarder() {
 			http_retry.WithBackoff(http_retry.BackoffLinear(retryTimeout)),
 		),
 	}
-	req := s.createRequest("GET", s.SimpleCtx())
+	req := s.createRequest("GET", s.SimpleCtx(), expectedContent)
 	resp, err := s.NewClient().Do(req)
 	require.NoError(s.T(), err, "call shouldn't fail")
 	assert.Equal(s.T(), failureCode, resp.StatusCode, "failure code should be propagated")
 	require.EqualValues(s.T(), 5, s.f.requestCount(), "backend should see all the retry calls")
 	s.ClientTripperware = clientTripperware
 }
+
+func (s *RetryTripperwareSuite) TestRequestBodyBuffering() {
+	s.f.resetFailingConfiguration(3, noSleep)
+	req := s.createRequest("GET", s.SimpleCtx(), expectedContent)
+	removeGetBody(req)
+	resp, err := s.NewClient().Do(req)
+	require.NoError(s.T(), err, "call shouldn't fail")
+	require.Equal(s.T(), httpwares_testing.DefaultPingBackStatusCode, resp.StatusCode, "response should succeed")
+	require.EqualValues(s.T(), 3, s.f.requestCount(), "3 requests should be retried to meet the modulo")
+}
+
+func (s *RetryTripperwareSuite) TestRequestBodyBufferingWithNoBody() {
+	s.f.resetFailingConfiguration(3, noSleep)
+	storedContent := expectedContent
+	expectedContent = ""
```

Comment:
I'm actually not sure what you're doing here. On the one hand you've added a parameter that looks like its supposed to replace the implicit usage of `expectedContent`, but you're also overwriting the value of `expectedContent`. Seems like you could do just one or the other?

Reply:
When we serve http requests in the test, they expect the request body to match 'expectedContent'. Here, we don't have a body so we need to set ' expectedContent = "" '. 'storedContent' then just stores the previous value so that it can be reset at the end of the test. This was the reason I initially introduced the other package-level variable. Do you think there's a better way of going about it?

Reply:
IIUC I don't think you need to add the parameter to `createRequest` - it can continue to use `expectedBody`

Reply:
Ah silly - agree, will update now

Reply:
Now updated